projects/aca-content/src/lib/store/effects/library.effects.ts

```diff
@@ -99,45 +101,59 @@ export class LibraryEffects {
       this.actions$.pipe(
         ofType<CreateLibraryAction>(LibraryActionTypes.Create),
         mergeMap(() => this.content.createLibrary()),
-        map((libraryId) => new NavigateLibraryAction(libraryId))
+        mergeMap((libraryId) => this.navigateToLibraryById(libraryId))
       ),
-    { dispatch: true }
+    { dispatch: false }
   );
 
   navigateLibrary$ = createEffect(
     () =>
       this.actions$.pipe(
         ofType<NavigateLibraryAction>(LibraryActionTypes.Navigate),
-        map((action) => {
-          const libraryId = action.payload;
-          if (libraryId) {
-            this.contentApi
-              .getNode(libraryId, { relativePath: '/documentLibrary' })
-              .pipe(map((node) => node.entry.id))
-              .subscribe(
-                (id) => {
-                  const route = action.route ? action.route : 'libraries';
-                  this.store.dispatch(new NavigateRouteAction([route, id]));
-                },
-                (error: HttpErrorResponse) => {
-                  switch (error.status) {
-                    case 403:
-                      this.notificationService.showWarning('APP.BROWSE.LIBRARIES.LIBRARY_NO_PERMISSIONS_WARNING');
-                      break;
-                    case 404:
-                      this.notificationService.showError('APP.BROWSE.LIBRARIES.ERRORS.LIBRARY_NOT_FOUND');
-                      break;
-                    default:
-                      this.notificationService.showError('APP.BROWSE.LIBRARIES.ERRORS.LIBRARY_LOADING_ERROR');
-                  }
+        mergeMap((action) => {
+          const payload = action.payload;
+          if (payload && 'guid' in payload) {
+            return this.store.select(isAdmin).pipe(
+              take(1),
+              mergeMap((isUserAdmin) => {
+                if (!isUserAdmin && payload.visibility !== 'PUBLIC' && !payload.role) {
+                  this.notificationService.showError('APP.BROWSE.LIBRARIES.LIBRARY_NO_PERMISSIONS_WARNING');
+                  return of(null);
```

Comment:
I see you have many return of(null) - here and also 127, 139 and 152 lines. I think you always return ```Observable<null>```. If there is no value and we always return null then can we just return nothing? Just have it as "void"? Similarly like above leaveLibrary action where we don't return anything.

Reply:
done